### PR TITLE
QOL Eggsac Carrier Changes - Sprites, Messages and Sounds

### PR DESCRIPTION
**Pull request by MPhonks**
# About the pull request

Adds new eggsac carrier sprites to the game. Includes eggsac overlay sprites that change based on how many eggs you carry. Implements a warning message for the hive when a carrier with eggs dies. Adds a sound effect to when an eggsac carrier dies. Fixes the egg drop chance upon death.

**Huge thanks to ihatethisengine for helping me with the coding.**

# Explain why it's good for the game

Eggsac Carrier currently has no sprites, there's no differentiating one from a regular carrier. 
This PR attempts to give more flavor to the strain. Also fixes something, so that's nice.

# Testing Photographs and Procedure (Updated)
<details>
<summary>Screenshots & Videos</summary>
 
 Sprites:
layers

Showcase Video:
https://github.com/cmss13-devs/cmss13/assets/134622054/e6f1f978-4ad7-41eb-8917-96440f8fd974

</details>

# Changelog

:cl: MarpleJones, ihatethisengine2
add: Added new sprites for the Eggsac Carrier. Includes an additional death sound for the eggsac bursting.
add: Added a hive announcement for when a Carrier dies with eggs.
fix: Carrier egg drop chance upon death now works as intended.
/:cl:

### CODE REVIEW

**Comment by MPhonks** [comment]
Changed back it back to being an open PR. >:)

File: code/modules/mob/living/carbon/xenomorph/castes/Carrier.dm [hunk 1]
```diff
@@ -145,10 +186,15 @@
...
-		while (eggs_cur > 0)
+		var/eggs_dropped = FALSE
+		for(var/i in 1 to eggs_cur)
 			if(prob(chance))
 				new /obj/item/xeno_egg(loc, hivenumber)
-				eggs_cur--
+				eggs_dropped = TRUE
+		eggs_cur = 0
+
+		if(eggs_dropped == TRUE) //Checks whether or not to announce egg drop.
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/castes/Carrier.dm:196]:
```suggestion
		if(eggs_dropped) //Checks whether or not to announce egg drop.
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Comment by MPhonks** [comment]
One last sprite change upcoming.

File: code/modules/mob/living/carbon/xenomorph/castes/Carrier.dm (hunk 1 above)
Comment (MPhonks) [reply code/modules/mob/living/carbon/xenomorph/castes/Carrier.dm]: Committed and fixed.

**Review by nauticall — APPROVED** [review]
sprites are really nice, approved 👍

**Pull request merged**